test-skripsi/test-prod.tsx

```diff
@@ -10,27 +17,33 @@
 
 	constructor() {
 		this.inventory = [];
+		this.loadInventory();
+	}
+
+	private findProductById(productId: number): Product | undefined {
+		return this.inventory.find((p) => p.id === productId);
 	}
 
-	addProduct(product: Product) {
-		const index = this.inventory.findIndex((p) => p.id === product.id);
-		if (index !== -1) {
-			console.log("Product already exists.");
-			return;
+	addProduct(product: Product): boolean {
+		if (this.findProductById(product.id)) {
```

Comment:
Throwing an `InventoryError` when a product already exists is a good practice. However, ensure that the calling code is properly handling these exceptions to avoid uncaught errors.